syntax = "proto3";

package gpt;

option go_package = "services/telegram-bot/lib";

message ErrorInfo {
  string code = 1;
  string message = 2;
}

service GptService {
  rpc ask (AskRequest) returns (AskResponse) {}
  rpc clearHistory (ClearHistoryRequest) returns (ClearHistoryResponse) {}
  rpc clearIndex (ClearIndexRequest) returns (ClearIndexResponse) {}
  rpc addGoogleDocs (AddGoogleDocsRequest) returns (AddGoogleDocsResponse) {}
}

message AskRequest {
  string query = 1;
  string conversationId = 2;
  string contextName = 3;
}

message AskResponse {
  string answer = 1;
  ErrorInfo error = 2;
}

message ClearHistoryRequest {
  string conversationId = 1;
}

message ClearHistoryResponse {}

message ClearIndexRequest {
  string contextName = 1;
}

message ClearIndexResponse {
  ErrorInfo error = 1;
}

message AddGoogleDocsRequest {
  string folderId = 1;
  string contextName = 2;
}

message AddGoogleDocsResponse {}
